### PR TITLE
Highlight Nodes on Network map

### PR DESCRIPTION
**Pull request by SourceDoctor**
Choose Node in Select Box and it's highlighted on Traffic Network Map

image

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/html/print-map.inc.php [hunk 1]
```diff
@@ -315,10 +334,25 @@
 $nodes = json_encode(array_values($devices_by_id));
 $edges = json_encode($links);
 
+array_multisort(array_column($devices_by_id, 'label'), SORT_ASC, $devices_by_id);
+
 if (count($devices_by_id) > 1 && count($links) > 0) {
 ?>
```
Comment (Jellyfrog) [inline includes/html/print-map.inc.php]: Maybe it should be a GET request so you can link to it

Comment (SourceDoctor) [reply includes/html/print-map.inc.php]: @Jellyfrog 
good idea, i'll change to it.
I'll adapt this to #10916 also, if it's merged :-)

Comment (SourceDoctor) [reply includes/html/print-map.inc.php]: @Jellyfrog 
on select box change URL not changing.
But on setting: `/map/highlight_node=<device_id>` Node is highlighted

File: includes/html/print-map.inc.php [hunk 2]
```diff
@@ -315,10 +334,25 @@
...
 
+<form name="printmapform" method="post" action="" class="form-horizontal" role="form">
```
Comment (Jellyfrog) [inline includes/html/print-map.inc.php]: Becuse this needs to be a GET?

Comment (SourceDoctor) [reply includes/html/print-map.inc.php]: if i change it to GET:
URL will be: `/map?highlight_node=<device_id>`. In classical PHP this is valid. Here it doesn't work.
It has to be changed to `/map/highlight_node=<device_id>` which works well

Comment (SourceDoctor) [reply includes/html/print-map.inc.php]: don't know what to do to get it to this here

Comment (SourceDoctor) [reply includes/html/print-map.inc.php]: @Jellyfrog some ideas?

Comment (SourceDoctor) [reply includes/html/print-map.inc.php]: is solved it with javascript. Maybe later a better way will be implemented :-)

**Comment by SourceDoctor** [comment]
@Jellyfrog something else to do here?

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-59-release-changelog-december-2019/10558/1

**Pull request merged**